File: generate_ppt.js
const pptxgen = require('pptxgenjs');
const path = require('path');
const os = require('os');
const fs = require('fs');

let pptx = new pptxgen();

// Define a pleasant master slide template
pptx.defineSlideMaster({
  title: 'MASTER_SLIDE',
  background: { fill: 'F8FAFC' }, // Soft, modern light grayish blue
  objects: [
    { rect: { x: 0, y: 0, w: '100%', h: 0.15, fill: { color: '3182CE' } } }, // Vibrant blue top bar
    { rect: { x: 0, y: 5.35, w: '100%', h: 0.3, fill: { color: 'E2E8F0' } } }, // Subtle gray footer
    { text: { text: 'Billware Pharmacy Software - Official Training Manual', options: { x: 0.5, y: 5.37, w: 5, h: 0.25, fontSize: 10, color: '718096' } } },
    { text: { text: 'Page | ', options: { x: 8.5, y: 5.37, w: 1, h: 0.25, fontSize: 10, color: '718096', align: 'right' } } }
  ],
  slideNumber: { x: 9.5, y: 5.37, color: '718096', fontSize: 10 }
});

const screenshotsDir = path.join(os.homedir(), 'Desktop', 'billware screenshots');
let imageFiles = [];
if (fs.existsSync(screenshotsDir)) {
    imageFiles = fs.readdirSync(screenshotsDir).filter(f => f.endsWith('.png') || f.endsWith('.jpg') || f.endsWith('.jpeg'));
    imageFiles.sort();
    imageFiles = imageFiles.filter(f => {
        const stats = fs.statSync(path.join(screenshotsDir, f));
        return stats.size > 1000;
    });
}

const slides_data = [
    { 
        title: "Billware - Smart Billing for Every Pharmacy", 
        content: [
            { text: "Welcome to the Billware Comprehensive User Manual.\n", options: { fontSize: 24, align: 'center', color: '2D3748', bold: true } },
            { text: "This guide will walk you through every feature of the platform in deep detail, from creating your account to managing daily sales, purchases, and advanced reporting features.", options: { fontSize: 16, align: 'center', color: '4A5568' } }
        ]
    },
    { 
        title: "1. Creating a New Store Account", 
        content: [
            { text: "Purpose: This is your gateway to setting up an independent, secure digital pharmacy store.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: From the login screen, click the 'Create an account' link. You will need to choose a unique Account Username. This username is permanent and serves as the primary identifier for your database. Next, create a strong, memorable password.", options: { bullet: true, color: '4A5568' } },
            { text: "Pro Tip: Each account operates in complete isolation. This means your inventory, sales, and customer data are 100% private and inaccessible to other store owners.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "2. Signing In to Your Dashboard", 
        content: [
            { text: "Purpose: Securely access your store's dashboard to begin your daily operations.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Enter the Username and Password you established during account creation. If you are using a secure, private computer (like the main counter PC), check the 'Keep me signed in' box. Click 'Sign In' to proceed.", options: { bullet: true, color: '4A5568' } },
            { text: "Security Note: Never check 'Keep me signed in' if you are logging in from a public or shared computer to protect your sensitive financial data.", options: { bullet: true, bold: true, color: 'E53E3E' } }
        ]
    },
    { 
        title: "3. Adding a New Customer", 
        content: [
            { text: "Purpose: Build a loyal customer database for faster billing and personalized service.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Navigate to Home > Add Customer. Enter the customer's Full Name, Phone Number, and their Address or Place of residence. Fields marked with a red asterisk (*) must be filled out before saving.", options: { bullet: true, color: '4A5568' } },
            { text: "Pro Tip: Always try to capture an accurate phone number. This allows you to easily pull up the customer's purchase history in the future and send digital invoices.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "4. Adding a New Doctor", 
        content: [
            { text: "Purpose: Maintain a registry of local prescribing doctors to attach to sales bills.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Go to Home > Add Doctor. Input the Doctor's Name, Contact Number, the Clinic or Hospital they are associated with, and their Address.", options: { bullet: true, color: '4A5568' } },
            { text: "Why This Matters: Registering doctors allows you to simply select their name from a dropdown menu during a busy checkout process, saving time and allowing you to track prescriptions.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "5. Adding a Purchase (Part 1 - Bill Details)", 
        content: [
            { text: "Purpose: Record new inventory arriving from your wholesale suppliers.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Under 'Bill Details', carefully type in the supplier's exact Bill Number and their Firm Name. Select the exact Date of purchase as printed on the physical invoice.", options: { bullet: true, color: '4A5568' } },
            { text: "Efficiency Tip: If your supplier provides digital invoices, use the 'Import Medicine List' feature. This will automatically populate the items below, saving you hours of manual typing.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "6. Adding a Purchase (Part 2 - Product Details)", 
        content: [
            { text: "Purpose: Enter specific line items into your inventory system.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: For each product, enter the Name, Batch Number, Expiry Date, Quantity, MRP, Rate, and GST percentage. Click '+ Add Another Item' to insert more rows.", options: { bullet: true, color: '4A5568' } },
            { text: "Crucial Step: The Batch Number and Expiry Date MUST be perfectly accurate. The entire Expiry Checker system relies on these two fields to alert you before medicines spoil.", options: { bullet: true, bold: true, color: 'E53E3E' } }
        ]
    },
    { 
        title: "7. Creating a New Sale Bill (Header Details)", 
        content: [
            { text: "Purpose: Initiate a new customer transaction at the checkout counter.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Open the Sales page. The Bill Number and Date are generated automatically. Use the dropdown menus to select an existing Customer and the Prescribing Doctor.", options: { bullet: true, color: '4A5568' } },
            { text: "Pro Tip: If a brand new customer walks in, you do not need to leave this page! Simply click the '+ New Customer' button to register them instantly without losing your current billing progress.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "8. Creating a New Sale Bill (Medicine Details)", 
        content: [
            { text: "Purpose: Add items to the customer's cart and finalize the sale.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Search for a medicine in the dropdown and select the specific Batch Number. Type in the Quantity (Qty) being sold. Review the Grand Total, collect payment, and click 'Save Bill'.", options: { bullet: true, color: '4A5568' } },
            { text: "Automation Magic: Selecting a Batch instantly pulls the Expiry Date, Current Stock, and MRP. Always verify your 'Current Stock' is sufficient before finalizing.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "9. Viewing Reports & Analytics", 
        content: [
            { text: "Purpose: Gain high-level insights into your pharmacy's financial performance.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Choose a 'From Date' and 'To Date'. Select the 'Type' of transaction you wish to review (Sales, Purchases, etc.) and hit 'Generate'.", options: { bullet: true, color: '4A5568' } },
            { text: "Business Intelligence: The four summary cards are your dashboard. The 'Net Profit' card is an instant health check, showing exactly what you've earned after purchase costs are deducted.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "10. Monitoring Medicine Stock", 
        content: [
            { text: "Purpose: Track exactly how much of every product you have on your shelves.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Click on 'Medicine Stock Report'. Search for items by Name or Batch. Look at 'Total In' vs 'Total Out' to see movement, and check 'Current Stock' for your physical inventory.", options: { bullet: true, color: '4A5568' } },
            { text: "Inventory Control: Use the quick filters ('Low Stock' and 'Out of Stock') daily. This creates an instant shopping list for your suppliers, ensuring you never turn a customer away.", options: { bullet: true, bold: true, color: 'E53E3E' } }
        ]
    },
    { 
        title: "11. Medicine Expiry Checker", 
        content: [
            { text: "Purpose: Prevent financial losses by identifying medicines before they expire.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Open the Expiry Checker. Select a predictive timeframe like 'This Month', '1 Month', or '2 Months' into the future, then click 'Check Now'.", options: { bullet: true, color: '4A5568' } },
            { text: "Proactive Management: By checking 2 months in advance, you have enough time to either return the stock to your distributor for a refund, or place it on discount to clear it out.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "12. Transaction Records", 
        content: [
            { text: "Purpose: Review a chronological audit log of everything happening in your store.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Browse through the unified list of Sales and Purchases. Use the top filters to isolate just Sales or just Purchases. Note the total values summarized at the top.", options: { bullet: true, color: '4A5568' } },
            { text: "Error Correction: If you realize a mistake was made on a previous bill, find the transaction here and click the red trash icon to delete it and return the stock to your inventory.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "13. Inbox & Notifications Overview", 
        content: [
            { text: "Purpose: A centralized hub for all system alerts, warnings, and messages.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Click the bell icon to open the Inbox. The 'All' tab shows a feed of every notification. Use 'Mark all read' to clear your queue once you have handled the issues.", options: { bullet: true, color: '4A5568' } },
            { text: "Visual Cues: Pay close attention to the colors. Red notifications demand immediate attention (like a critical stockout), while Orange acts as a warning for future events.", options: { bullet: true, bold: true, color: 'E53E3E' } }
        ]
    },
    { 
        title: "14. Managing Medicine Alerts", 
        content: [
            { text: "Purpose: A dedicated channel just for inventory-related urgencies.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Navigate to the 'Medicines' tab within the Inbox. This filters out general business news and only shows you items that are 'Expiring Soon' or 'Out of Stock'.", options: { bullet: true, color: '4A5568' } },
            { text: "Daily Routine: Make it a habit to check this specific tab first thing in the morning. It acts as your daily to-do list for restocking the shelves and pulling expired boxes.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "15. Tracking Business Updates", 
        content: [
            { text: "Purpose: Stay motivated and informed about your store's operational milestones.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Switch to the 'Business' tab in the Inbox. Here you will see congratulatory messages for hitting sales targets, summaries of daily performance, and operational alerts.", options: { bullet: true, color: '4A5568' } },
            { text: "Why It Matters: These alerts help you keep your finger on the pulse of the business without having to dig through dense reports and spreadsheets.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "16. System Updates & Announcements", 
        content: [
            { text: "Purpose: Communication directly from the Billware software team.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Check the 'Updates' tab periodically. This is where you will read release notes, discover new features that have been added, and learn about any scheduled maintenance.", options: { bullet: true, color: '4A5568' } },
            { text: "Pro Tip: Reading these updates ensures you are getting the most value out of Billware by utilizing the latest tools designed to make your life easier.", options: { bullet: true, bold: true, color: '3182CE' } }
        ]
    },
    { 
        title: "17. Managing Shop Settings", 
        content: [
            { text: "Purpose: Configure the legal and visual identity of your pharmacy.", options: { bullet: true, bold: true, color: '2D3748' } },
            { text: "Key Actions: Navigate to Settings. Input your Store Name, Address, and Contact numbers. Add legal details like your Drug Licence No and GSTIN. Customize the footer message for receipts.", options: { bullet: true, color: '4A5568' } },
            { text: "Compliance Warning: The information saved here is printed on every single invoice you hand to a customer. It is a strict legal requirement that your GSTIN and Drug License numbers are completely accurate.", options: { bullet: true, bold: true, color: 'E53E3E' } }
        ]
    }
];

// Map images to slides
for (let i = 1; i < slides_data.length; i++) {
    if (i - 1 < imageFiles.length) {
        slides_data[i].screenshotPath = path.join(screenshotsDir, imageFiles[i-1]);
    }
}

// Set up master slide layout
pptx.layout = 'LAYOUT_16x9';

// Title Slide
let titleSlide = pptx.addSlide({ masterName: 'MASTER_SLIDE' });
titleSlide.addText(slides_data[0].title, { x: 0.5, y: 1.8, w: 9, h: 1, fontSize: 36, bold: true, align: 'center', color: '1A365D' });
titleSlide.addText(slides_data[0].content, { x: 0.5, y: 3.0, w: 9, h: 1.5 });

// Content Slides
for (let i = 1; i < slides_data.length; i++) {
    let s = slides_data[i];
    let newSlide = pptx.addSlide({ masterName: 'MASTER_SLIDE' });
    
    // Add title
    newSlide.addText(s.title, { x: 0.5, y: 0.25, w: 9, h: 0.4, fontSize: 24, bold: true, color: '2B6CB0', align: 'left' });
    
    // Add Image perfectly aligned
    if (s.screenshotPath && fs.existsSync(s.screenshotPath)) {
        newSlide.addImage({ path: s.screenshotPath, x: 0.5, y: 0.75, w: 9, h: 3.3, sizing: {type: 'contain'} });
    } else {
        newSlide.addShape(pptx.ShapeType.rect, { x: 0.5, y: 0.75, w: 9, h: 3.3, fill: { color: 'E2E8F0' } });
        newSlide.addText("[ Missing Screenshot ]", { x: 0.5, y: 0.75, w: 9, h: 3.3, fontSize: 18, align: 'center', color: 'A0AEC0' });
    }
    
    // Add Instructions text aligned perfectly below the image
    newSlide.addText(s.content, { 
        x: 0.4, 
        y: 4.15, 
        w: 9.2, 
        h: 1.1, 
        fontSize: 11.5, 
        color: '363636', 
        valign: 'top',
        align: 'left',
        margin: 5,
        lineSpacing: 14
    });
}

const desktopPath = path.join(os.homedir(), 'Desktop', 'Billware_User_Instructions_Pleasant.pptx');

pptx.writeFile({ fileName: desktopPath })
    .then(fileName => {
        console.log(`Created file: ${fileName}`);
    })
    .catch(err => {
        console.error(err);
    });

File: generate_ppt.py
from pptx import Presentation
from pptx.util import Inches, Pt
from pptx.enum.text import PP_ALIGN
from pptx.dml.color import RGBColor
import os

try:
    prs = Presentation()

    slides_data = [
        {
            "title": "Billware - Smart Billing for Every Pharmacy",
            "content": "User Manual & Instructions",
            "screenshot": None
        },
        {
            "title": "1. Creating a New Store Account",
            "content": (
                "1. Open the Billware application.\n"
                "2. Click on the 'Create an account' link on the sign-in page.\n"
                "3. Enter your desired Account Username.\n"
                "4. Create a strong Password and re-enter it to confirm.\n"
                "5. Click the 'Create Account' button.\n\n"
                "Note: Each account operates as a completely independent, private store."
            ),
            "screenshot": "Screenshot 1: Create Account Page"
        },
        {
            "title": "2. Signing In to Your Dashboard",
            "content": (
                "1. Enter your Username and Password.\n"
                "2. Optional: Check 'Keep me signed in on this device' for faster access.\n"
                "3. Click the 'Sign In' button to access your dashboard.\n\n"
                "Tip: If you forgot your password, contact the pharmacy owner."
            ),
            "screenshot": "Screenshot 2: Sign In Page"
        },
        {
            "title": "3. Adding a New Customer",
            "content": (
                "1. Navigate to Home > Add Customer.\n"
                "2. Enter the customer's Name, Phone Number, and Address / Place. Mandatory fields are marked with a red asterisk (*).\n"
                "3. Click 'Save Customer' to add them to your database, or 'Cancel' to discard."
            ),
            "screenshot": "Screenshot 3: Add New Customer Page"
        },
        {
            "title": "4. Adding a New Doctor",
            "content": (
                "1. Navigate to Home > Add Doctor.\n"
                "2. Enter the Doctor Name, Phone Number, Clinic / Hospital Name, and Address / Place.\n"
                "3. Click 'Save Doctor' to register the doctor for future billing references."
            ),
            "screenshot": "Screenshot 4: Add New Doctor Page"
        },
        {
            "title": "5. Adding a Purchase (Part 1 - Bill Details)",
            "content": (
                "1. Navigate to Home > Purchase.\n"
                "2. Under Bill Details, enter the supplier's Bill No and the Firm (supplier name).\n"
                "3. Select the Date of purchase.\n"
                "4. Use 'Import Medicine List' to quickly load items, or clear them using 'Clear Imported'."
            ),
            "screenshot": "Screenshot 5: Add Purchase - Top"
        },
        {
            "title": "6. Adding a Purchase (Part 2 - Product Details)",
            "content": (
                "1. For each item, enter the Product Name, Batch No, Expire Date, Pack size, Quantity, MRP, Rate, and GST (%).\n"
                "2. The Total will be auto-calculated.\n"
                "3. Click '+ Add Another Item' if you have more products on the same bill.\n"
                "4. Review the Grand Total at the bottom and click 'Save All Purchases'."
            ),
            "screenshot": "Screenshot 6: Add Purchase - Bottom"
        },
        {
            "title": "7. Creating a New Sale Bill (Header Details)",
            "content": (
                "1. Navigate to Home > Sales. The Bill No and Date are auto-filled.\n"
                "2. Select an existing Customer Name from the dropdown, or click '+ New Customer' to add one.\n"
                "3. Select an existing Doctor Name, or click '+ New Doctor'."
            ),
            "screenshot": "Screenshot 7: New Sale Bill - Top"
        },
        {
            "title": "8. Creating a New Sale Bill (Medicine Details)",
            "content": (
                "1. In the Medicine Details section, select a product from the dropdown and choose the Batch No.\n"
                "2. The system auto-fills Exp Date, Pack, Stock, MRP, etc.\n"
                "3. Enter the Qty (quantity) you are selling.\n"
                "4. Click '+ Add Medicine Row' to add up to 15 items per bill.\n"
                "5. Check the Grand Total and click 'Save Bill' to finalize the sale."
            ),
            "screenshot": "Screenshot 8: New Sale Bill - Bottom"
        },
        {
            "title": "9. Viewing Reports & Analytics",
            "content": (
                "1. Navigate to Home > Report.\n"
                "2. Select a From Date and To Date, and choose the Type of transactions. Click 'Generate'.\n"
                "3. View summary cards: Total Sales, Total Purchase, Net Profit, and Total Items.\n"
                "4. Use Action Buttons to Print Report, Backup, or Restore data.\n"
                "5. Use 'Search Past Bills' to find historical bills."
            ),
            "screenshot": "Screenshot 9: Report Dashboard"
        },
        {
            "title": "10. Monitoring Medicine Stock",
            "content": (
                "1. On the Report page, click on 'Medicine Stock Report' to expand it.\n"
                "2. Use the search bar to find specific medicines by name or batch.\n"
                "3. Filter the view by clicking 'In Stock', 'Low Stock', 'Out of Stock', or 'Total Items'.\n"
                "4. Monitor the Total In, Total Out, and Current Stock columns to track inventory health."
            ),
            "screenshot": "Screenshot 10: Medicine Stock Report"
        },
        {
            "title": "11. Medicine Expiry Checker",
            "content": (
                "1. Navigate to the Expiry Checker tool.\n"
                "2. Select a timeframe to check (e.g., This Month, 1 Month, 2 Months, or Custom).\n"
                "3. Click 'Check Now' to generate the report.\n"
                "4. View the summary tabs and detailed list below to manage stock before it expires."
            ),
            "screenshot": "Screenshot 11: Medicine Expiry Checker"
        },
        {
            "title": "12. Transaction Records",
            "content": (
                "1. Navigate to Transaction Records to view a detailed log of all activities.\n"
                "2. Use the filters at the top to toggle between All, Sales, and Purchase records.\n"
                "3. View the summary statistics for Records, Quantity, Sales, Purchase, and Net amount.\n"
                "4. Use the red delete icon on the right to remove a specific record if needed."
            ),
            "screenshot": "Screenshot 12: Transaction Records"
        },
        {
            "title": "13. Inbox & Notifications Overview",
            "content": (
                "1. Click the Inbox icon in the top right to view alerts and notifications.\n"
                "2. The 'All' tab shows a chronological feed of all alerts.\n"
                "3. Use the 'Mark all read' button to clear the notifications.\n"
                "4. Notifications are color-coded (red for urgent, orange for warnings)."
            ),
            "screenshot": "Screenshot 13: Inbox panel - All"
        },
        {
            "title": "14. Managing Medicine Alerts",
            "content": (
                "1. Inside the Inbox, click the 'Medicines' tab to filter out business alerts.\n"
                "2. Quickly spot medicines that are Out of Stock or Expiring Soon.\n"
                "3. Use this view for daily inventory restocking and removing expiring items."
            ),
            "screenshot": "Screenshot 14: Inbox - Medicines tab"
        },
        {
            "title": "15. Tracking Business Updates",
            "content": (
                "1. Inside the Inbox, click the 'Business' tab to view operational alerts.\n"
                "2. Track milestones like total daily sales and performance metrics.\n"
                "3. Stay updated on store activity without leaving your current screen."
            ),
            "screenshot": "Screenshot 15: Inbox - Business tab"
        },
        {
            "title": "16. System Updates & Announcements",
            "content": (
                "1. Inside the Inbox, click the 'Updates' tab to view platform announcements.\n"
                "2. Read about new features, system maintenance, or important changes to Billware.\n"
                "3. Keep up to date with the latest tools designed to help your pharmacy."
            ),
            "screenshot": "Screenshot 16 & 17: Inbox - Updates"
        },
        {
            "title": "17. Managing Shop Settings",
            "content": (
                "1. Navigate to the Shop Settings page to manage your store identity.\n"
                "2. Review your Store Name, Invoice Prefix, Phone Number, and Address.\n"
                "3. Update Legal Information like Drug Licence No and GSTIN.\n"
                "4. Customize Invoice Terms & Conditions and Footer Message.\n"
                "5. Note: If editing is locked, click 'Request Edit Permission' to ask the Super Admin for access."
            ),
            "screenshot": "Screenshot 18: Shop Settings"
        }
    ]

    # Title Slide
    title_slide_layout = prs.slide_layouts[0]
    slide = prs.slides.add_slide(title_slide_layout)
    title = slide.shapes.title
    subtitle = slide.placeholders[1]
    title.text = slides_data[0]["title"]
    subtitle.text = slides_data[0]["content"]

    # Content Slides
    bullet_slide_layout = prs.slide_layouts[5] # Title only

    for slide_data in slides_data[1:]:
        slide = prs.slides.add_slide(bullet_slide_layout)
        shapes = slide.shapes
        
        title_shape = shapes.title
        title_shape.text = slide_data["title"]
        
        # Add a placeholder box for the screenshot
        left = Inches(0.5)
        top = Inches(1.2)
        width = Inches(9)
        height = Inches(3.8)
        shape = shapes.add_shape(
            1, left, top, width, height # 1 is MSO_SHAPE.RECTANGLE
        )
        shape.fill.solid()
        shape.fill.fore_color.rgb = RGBColor(220, 220, 220)
        text_frame = shape.text_frame
        text_frame.text = f"[ Please Replace This Box With {slide_data['screenshot']} ]"
        text_frame.paragraphs[0].alignment = PP_ALIGN.CENTER
        
        # Add text box for instructions
        left = Inches(0.5)
        top = Inches(5.2)
        width = Inches(9)
        height = Inches(2.0)
        txBox = shapes.add_textbox(left, top, width, height)
        tf = txBox.text_frame
        tf.word_wrap = True
        
        p = tf.add_paragraph()
        p.text = slide_data["content"]
        p.font.size = Pt(14)

    output_path = os.path.join(os.getcwd(), 'Billware_User_Instructions.pptx')
    prs.save(output_path)
    print(f"Success! Presentation saved to: {output_path}")

except Exception as e:
    print(f"Error generating PPT: {str(e)}")
